bpaf_core/src/split.rs

```diff
@@ -185,15 +185,19 @@ pub(crate) enum Arg<'a> {
     },
 }
 
-fn ascii_prefix(input: &OsStr) -> Option<(Cow<str>, Cow<OsStr>)> {
+fn ascii_prefix(input: &OsStr) -> Option<(Cow<str>, Option<Cow<OsStr>>)> {
```

Comment:
My idea for `OsOrStr` type was that it can represent both `OsString` and `String` and if it can be represented as a regular string - that's what we are going to use. I guess those two implementation needs to be updated for that:

```rust
impl<'a> From<&'a OsStr> for OsOrStr<'a> {
    fn from(value: &'a OsStr) -> Self {
        Self::Os(Cow::Borrowed(value))
    }
}

impl From<OsString> for OsOrStr<'static> {
    fn from(value: OsString) -> Self {
        Self::Os(Cow::Owned(value))
    }
}
```

If this holds - `ascii_prefix`, if ascii prefix is present, should always produce a non empty prefix and non empty suffix.

In production the input to the parser is going to be `OsString`, but most of them are going to be valid utf8 - hence the representation that allows to use `OsString` only when we must.